src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -32,56 +32,82 @@ namespace Microsoft.Diagnostics.Tracing.Analysis
     /// </summary>
     public static class TraceLoadedDotNetRuntimeExtensions
     {
+        internal static readonly string LoadedDotNetRuntimesKey = "Computers/LoadedDotNetRuntimes";
+
         public static void NeedLoadedDotNetRuntimes(this TraceEventDispatcher source)
         {
+            if (source.userData.ContainsKey(LoadedDotNetRuntimesKey))
+            {
+                // The callbacks are already registered for this source
+                return;
+            }
+
             // ensure there are base processes
```

Comment:
💭 From the comments as a whole, it sounds like what you are saying is `NeedLoadedDotNetRuntimes` should behave as follows:

* The first time it's called, sets up callbacks and initializes the dictionary for information
* If called again later, leaves the callbacks in place but otherwise performs the same initialization - clearing the existing dictionary

If this is how you would describe this method, I should be able to implement/document/test this behavior in a straightforward manner. But I'll wait to hear back before proceeding down this path.